static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js

```diff
@@ -0,0 +1,100 @@
+define([
+    'bean',
+    'bonzo',
+    'fastdom',
+    'common/utils/_',
+    'common/utils/$',
+    'common/utils/detect',
+    'common/utils/mediator',
+    'common/utils/storage',
+    'common/utils/template',
+    'common/views/svgs',
+    'text!common/views/commercial/creatives/expandable-video-v2.html'
+], function (
+    bean,
+    bonzo,
+    fastdom,
+    _,
+    $,
+	detect,
+    mediator,
+    storage,
+    template,
+    svgs,
+    ExpandableVideoTpl
+) {
+
+    /**
+     * https://www.google.com/dfp/59666047#delivery/CreateCreativeTemplate/creativeTemplateId=10028247
+     */
+    var ExpandableVideo = function ($adSlot, params) {
+        this.$adSlot      = $adSlot;
+        this.params       = params;
+        this.isClosed     = true;
+
+        if (detect.isBreakpoint({min: 'tablet'})) {
+            this.closedHeight = 250;
+            this.openedHeight = 500;
+        } else {
+            this.closedHeight = 150;
+            this.openedHeight = 300;
+        }
+
+    };
+
+    ExpandableVideo.prototype.create = function () {
+        var videoHeight = this.openedHeight,
+            showmoreArrow = {
+                showArrow: (this.params.showMoreType === 'arrow-only' || this.params.showMoreType === 'plus-and-arrow') ?
+                    '<button class="ad-exp__open-chevron ad-exp__open">' + svgs('arrowdownicon') + '</button>' : ''
+            },
+            showmorePlus = {
+                showPlus: (this.params.showMoreType === 'plus-only' || this.params.showMoreType === 'plus-and-arrow') ?
+                    '<button class="ad-exp__close-button ad-exp__open">' + svgs('closeCentralIcon') + '</button>' : ''
+            },
+            videoSource = {
+                videoEmbed: (this.params.YoutubeVideoURL !== '') ?
+                    '<iframe id="YTPlayer" width="100%" height="' + videoHeight + '" src="' + this.params.YoutubeVideoURL + '?showinfo=0&amp;rel=0&amp;controls=0&amp;fs=0&amp;title=0&amp;byline=0&amp;portrait=0" frameborder="0" class="expandable-video"></iframe>' : ''
+            },
+            $ExpandableVideo = $.create(template(ExpandableVideoTpl, { data: _.merge(this.params, showmoreArrow, showmorePlus, videoSource) })),
+            domPromise = new Promise(function (resolve) {
```

Comment:
Again, not your style - ours - but should we be using fastdom by default? Using fastdom for things we haven't profiled means adding a lot of boilerplate to deal with asynchronicity. It also makes debugging trickier. I guess you could argue that all these DOM operations add up, so using fastdom as our first choice heads those problems off right from the start, but still... it seems to make DOM code look a lot more complicated than its actual behaviour really warrants. 

Then again, it's late... perhaps I'll see things differently tomorrow morning!